core/domain/stats_services_test.py

```diff
@@ -151,6 +151,118 @@ def test_calls_to_stats_methods(self):
+            'cmd': exp_domain.CMD_ADD_STATE,
+            'state_name': 'New state'
+        })]
+        exp_services.update_exploration(
+            'committer_id_v3', exploration.id, change_list, 'Added new state')
+        exploration = exp_services.get_exploration_by_id(exp_id)
+        exp_issues = stats_services.get_exp_issues(exp_id, exploration.version)
+        self.assertEqual(exp_issues.exp_version, 2)
+        self.assertEqual(exp_issues.unresolved_issues, [])
+
+        # Create a playthrough and assign it to an issue in exploration issues
+        # model.
+        playthrough_id1 = stats_models.PlaythroughModel.create(
+            exploration.id, exploration.version, 'EarlyQuit', {
+                'state_name': {
+                    'value': 'New state'
+                },
+                'time_spent_in_exp_in_msecs': {
+                    'value': 200
+                }
+            }, [{
+                'action_type': 'ExplorationStart',
+                'action_customization_args': {
+                    'state_name': {
+                        'value': 'New state'
+                    }
+                },
+                'schema_version': 1
+            }])
+        exp_issue1 = stats_domain.ExplorationIssue.from_dict({
+            'issue_type': 'EarlyQuit',
+            'issue_customization_args': {
+                'state_name': {
+                    'value': 'New state'
+                },
+                'time_spent_in_exp_in_msecs': {
+                    'value': 200
+                }
+            },
+            'playthrough_ids': [playthrough_id1],
+            'schema_version': 1,
+            'is_valid': True
+        })
+        exp_issues.unresolved_issues = [exp_issue1]
+        stats_services.save_exp_issues_model_transactional(exp_issues)
+
+        # Delete a state.
+        change_list = [exp_domain.ExplorationChange({
+            'cmd': exp_domain.CMD_DELETE_STATE,
+            'state_name': 'New state'
+        })]
+        exp_services.update_exploration(
+            'committer_id_v3', exploration.id, change_list, 'Deleted a state')
+        exploration = exp_services.get_exploration_by_id(exp_id)
+        exp_issues = stats_services.get_exp_issues(exp_id, exploration.version)
+        self.assertEqual(exp_issues.exp_version, 3)
+        self.assertEqual(exp_issues.unresolved_issues[0].to_dict(), {
+            'issue_type': 'EarlyQuit',
+            'issue_customization_args': {
+                'state_name': {
+                    'value': 'New state'
+                },
+                'time_spent_in_exp_in_msecs': {
+                    'value': 200
+                }
+            },
+            'playthrough_ids': [playthrough_id1],
+            'schema_version': 1,
+            'is_valid': False
+        })
+
+        # Revert to an older version, exploration issues model also changes.
+        exp_services.revert_exploration('committer_id_v4', exp_id, 3, 2)
```

Comment:
nit: use kwargs here to document these ints.

Reply:
Done.